py-polars/polars/_plr.py

```diff
@@ -2,13 +2,81 @@
 # bit of trickery here to allow overwriting it with other function pointers.
 
 import builtins
+import os
 import sys
 
+PKG_VERSION = "1.33.1"
+
+
+def pllts() -> None:
+    import _polars_lts_cpu._polars_lts_cpu as plr
+
+    sys.modules[__name__] = plr
+
+
+def pl64() -> None:
+    import _polars64._polars64 as plr
+
+    sys.modules[__name__] = plr
+
+
+def pl32() -> None:
+    import _polars32._polars32 as plr
+
+    sys.modules[__name__] = plr
+
+
 if hasattr(builtins, "__POLARS_PLR"):
     sys.modules[__name__] = builtins.__POLARS_PLR
 else:
-    # PyO3 registers a module called `polars` in the polars namespace that contains all
-    # the functions that we want to have exposed.
-    import polars.polars as plr
+    # Each of the Polars variants registers a `_polars...` package that we can import
+    # the PLR from.
 
-    sys.modules[__name__] = plr
+    _force = os.environ.get("POLARS_FORCE_PKG")
+    _prefer = os.environ.get("POLARS_PREFER_PKG")
+
+    pkgs = {"lts": pllts, "64": pl64, "32": pl32}
+
+    if _force is not None:
+        try:
+            pkgs[_force]()
+
+            if sys.modules[__name__].__version__ != PKG_VERSION:
+                msg = f"Polars Rust module for '{_force}' ({sys.modules[__name__].__version__}) did not match version of Python package '{PKG_VERSION}'"
+                raise ImportError(msg)
+        except KeyError:
+            msg = f"Invalid value for `POLARS_FORCE_PKG` variable: '{_force}'"
+            raise ValueError(msg) from None
+    else:
+        preference = [pllts, pl64, pl32]
+        if _prefer is not None:
+            try:
+                preference.insert(0, pkgs[_prefer])
+            except KeyError:
+                msg = f"Invalid value for `POLARS_PREFER_PKG` variable: '{_prefer}'"
+                raise ValueError(msg) from None
+
+        for pkg in preference:
+            try:
+                pkg()
+
+                if sys.modules[__name__].__version__ != PKG_VERSION:
+                    import warnings
+
+                    warnings.warn(
+                        f"Skipping Polars' Rust module version '{sys.modules[__name__].__version__}' did not match version of Python package '{PKG_VERSION}'.",
+                        ImportWarning,
+                        stacklevel=2,
+                    )
+                    continue
+
+                break
+            except ImportError:
+                pass
+        else:
+            msg = "could not find Polars' Rust module"
```

Comment:
Can we store a list of all emitted warnings, and if we reach this point, include them into the error? Because right now if you get a version mismatch you'll just get this message which can be rather confusing/unclear.